src/assignments/tech-jobs-oo.rst

```diff
@@ -1,6 +1,586 @@
+and the other non-ID fields are no longer strings. Instead, they are classes of
+their own.
+
+Job Fields
+^^^^^^^^^^^
+
+Open the ``Job`` class file. You’ll see the following fields (among others):
+
+.. sourcecode:: java
+   :linenos:
+
+   private String name;
+   private Employer employer;
+   private Location location;
+   private PositionType positionType;
+   private CoreCompetency coreCompetency;
+
+Of these, only ``name`` is a string. Sally created classes to represent each of
+the other properties. These classes---``Employer``, ``Location``,
+``CoreCompetency``, ``PositionType``---have ``value`` and ``id`` fields.
+
+So, for example, if you had a ``Job`` instance, you could get the name of the
+employer this way:
+
+.. sourcecode:: java
+
+   // job is an instance of Job
+   String employerName = job.getEmployer().getValue();
+
+Additionally, the ``toString()`` method of the ``Employer`` class is set up to
+return the ``value`` field. Thus, using one of these objects in another string
+context like ``System.out.println`` will print the data stored in ``value``.
+
+.. sourcecode:: java
+
+   // Prints the name of the employer
+   System.out.println(job.getEmployer);
+
+Why do we go to all of this trouble when we could store this job-related data
+as strings? There are a couple of reasons.
+
+Eliminate Duplication of Data
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In our app, we have multiple jobs that have the same value in a given field.
+For example, there are multiple jobs with position type “Web - Full Stack”, and
+there are different jobs with the same employer. If we store the values of
+these fields as strings directly within each ``Job`` object, that data would be
+repeated in several places across the application.
+
+By using objects, we can have a single ``PositionType`` object with value “Web
+- Full Stack”. Each job that wants to use that position type holds onto a
+reference to the given object. Similarly, we can have one ``Employer`` object
+for each employer.
+
+Aside from reducing the amount of raw data / memory that the application uses,
+this will allow data to be updated more easily and properly. If we need to
+change the name of an employer (e.g. due to a typo or a name change at the
+company), we can change it in one place---the single ``Employer`` object that
+represents that company.
+
+Enable Extension
+~~~~~~~~~~~~~~~~~
+
+While the four ``Job`` properties represented by objects will primarily be used
+for their string values, it’s easy to imagine adding new properties to address
+future needs.
+
+For example, it would be useful for an ``Employer`` object to have an address,
+a primary contact, and a list of jobs available at that employer.
+
+For a ``Location`` object, useful information includes a list of zip codes
+associated with that location, in order to determine the city and state for an
+employer or job.
+
+If we were to store these properties as strings, extending and modifying this
```

Comment:
"These properties" is ambiguous in this sentence, since in the previous sentence you are talking about properties that might be on a `Location` object.

Reply:
I rephrased the paragraph.  Let me know what you think.